Mathlib/GroupTheory/MonoidLocalization.lean

```diff
@@ -1886,6 +1907,80 @@ noncomputable def lift (f : LocalizationWithZeroMap S N) (g : M →*₀ P)
       exact f.toMonoidWithZeroHom.map_zero.symm }
 #align submonoid.localization_with_zero_map.lift Submonoid.LocalizationWithZeroMap.lift
 
+/-- Given a Localization map `f : M →*₀ N` for a Submonoid `S ⊆ M`,
+`f 0 * (f y)⁻¹ = 0` for any `y : M` -/
+theorem eq_zero_of_fst_eq_zero (f : LocalizationWithZeroMap S N)
+    {z x} {y : S} (h : z * f.toFun y = f.toFun x)
+  (hx : x = 0) : z = 0 := by
+  rw [hx, f.map_zero'] at h
+  exact (IsUnit.mul_left_eq_zero (LocalizationMap.map_units' f.toLocalizationMap y)).1 h
```

Comment:
This is a weird statement. The docstring doesn't seem to correspond to the Lean statement, and I would have expected something like
```suggestion
/-- Given a Localization map `f : M →*₀ N` for a Submonoid `S ⊆ M`,
`f 0 * (f y)⁻¹ = 0` for any `y : M` -/
theorem eq_zero_of_fst_eq_zero (f : LocalizationWithZeroMap S N)
    {z} {y : S} : z * f y = 0 ↔ z = 0 := (f.toLocalizationMap.map_units' y).mul_left_eq_zero
```
which shows you that probably you don't need that lemma at all

Reply:
The reason I put this lemma is because it exists a version for IsLocalization for Semirings in [IsLocalization.eq_zero_of_fst_eq_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Localization/Basic.html#IsLocalization.eq_zero_of_fst_eq_zero). But I agree when I was trying to describe I was not sure about it...

Reply:
Meh, honestly I wouldn't bother about it (and would even remove `IsLocalization.eq_zero_of_fst_eq_zero`)

Reply:
I removed that result and modified the proof of the next result accordingly. I am not very happy about the way I wrote it: because I don't know how to write it in a short way, may be since some inferences (?) are not done automatically... Anyway, I hope the proof is clear.